model_viewer/hob_mesh.pas

```diff
@@ -351,32 +360,37 @@ procedure TModel.DrawGL(var opts: TRenderOpts);
   end;
 
   glColor3f(1, 1, 1);
-
-  if opts.fg_all then begin
-      for k := 0 to Length(_triangles) - 1 do begin
-          if _triangles[k].Size > 0 then
-              for i := 0 to _triangles[k].Size - 1 do
-                  DrawTri(_triangles[k][i]);
+  if _triangles.Size > 0 then begin

```

Comment:
Use ```  if _triangles.Size = 0 then
      exit; ``` at the start of the function. Does this actually happen with any file?

Reply:
Yes, this happens with the opkg2_HOB files per level.